generators/client/templates/angular/src/main/webapp/app/_app.module.ts

```diff
@@ -73,9 +82,74 @@ import {
     ],
     providers: [
         ProfileService,
-        customHttpProvider(),
         PaginationConfig,
-        UserRouteAccessService
+        UserRouteAccessService,
+        <%_ if (authenticationType === 'jwt') { _%>
```

Comment:
this template seems wrong compared to existing logic, correct one should be
```
        <%_ if (authenticationType === 'jwt') { _%>
        {
             provide: HTTP_INTERCEPTORS,
             useClass: AuthInterceptor,
             multi: true,
             deps: [
                 LocalStorageService,
                 SessionStorageService
             ]
         },
         <%_ } _%>
         {
             provide: HTTP_INTERCEPTORS,
             useClass: AuthExpiredInterceptor,
             multi: true,
             deps: [
                 Injector<% if (['session', 'oauth2'].includes(authenticationType)) { %>,
                 StateStorageService<% } if(authenticationType === 'session') %>,
                 LoginModalService<% } %>
             ]
         },
         {
             provide: HTTP_INTERCEPTORS,
             useClass: ErrorHandlerInterceptor,
             multi: true,
             deps: [
                 JhiEventManager
             ]
         },
         {
             provide: HTTP_INTERCEPTORS,
             useClass: NotificationInterceptor,
             multi: true,
             deps: [
                 Injector
             ]
         }
```

Reply:
we can shrink them to only switch the dependencies i removed the Modal to solve cyclic dependencies in case of session based auth

Reply:
merged them